Also retrieve class side methods in class search

Olliwehr opened the pull request:
* When searching inside a class (by navigating into it), now class methods are shown too.
* For clarification those results get a '(class)' prefix.
* closes #9

Commit:
Also retrieve class side methods in class search

When searching inside a class (by navigating into it), now class methods are shown too.
For clarification those results get a '(class)' prefix.

Olliwehr commented on packages/Algernon.package/ALGDynamicWorldMenuType.class/methodProperties.json:
```diff
@@ -1,8 +1,6 @@
 		"extractMenuEntriesFrom:" : "NL 6/29/2021 15:00",
-		"filterOutDuplicates:" : "NM 6/29/2021 17:28",
-		"menuEntries:" : "NL 6/4/2021 16:21" },
+		"filterOutDuplicates:" : "NM 6/29/2021 17:28" },
```
Really don't know how these changes got in here. As they defintively belong to prior merged stuff, maybe this was not merged correctly? 🤔
But it didn't breakt sth here, soo... 😅

Commit:
Make actions work on class methods in class search

By introducing the '(class)' prefix, actions for this item didn't work anymore.
In those cases the prefix gets ignored now to make it work again.

Commit:
Will this work in CI?

niklasmohrin commented on packages/Algernon.package/ALGMethodType.class/class/algertemListWith..st:
```diff
@@ -1,4 +1,10 @@
+		collect: [:each | ALGItem withAlgertype: (self withType: each withClass: aClass) ].
+		
+	items addAll: (aClass class selectors asOrderedCollection
+		collect: [:each | ALGItem withAlgertype: (self withType: self prefixForClassMethods, each withClass: aClass class) ]).
```
> `aClass class`

Wow, that is odd :laughing: (but correct)

Olliwehr replied:
Yeah, quite strange 😄 But as far as my understanding goes `aClass class` then refers to the meta-class of `aClass` (which internally handles the method dictionary for the class side of `aClass`).

niklasmohrin commented on packages/Algernon.package/ALGMethodType.class/instance/createListOfClassAwareAlgeractions..st:
```diff
@@ -0,0 +1,5 @@
+createListOfClassAwareAlgeractions: aMethod
+	methodClass
+		ifNil: [^ {}]
+		ifNotNil: [^ {self createActionBrowseExistingFor: aMethod. self createActionBrowseNewFor: aMethod}]
```
nitpick: the return can be moved outside of the branches (also missing `self`; formatting up for debate):

```suggestion
^ self methodClass
    ifNil: [ {} ]
    ifNotNil: [ {
        self createActionBrowseExistingFor: aMethod .
        self createActionBrowseNewFor: aMethod        } ]
```

Olliwehr replied:
@niklasmohrin i would directly commit your suggestion from here, but it seems like the method body then lacks one indent or is that caused by the view in github?

niklasmohrin replied:
oh no, I didn't write that properly :D

Commit:
Apply Niklas' suggestion

Commit:
Do some refactoring on the tests for the class method stuff

As i noticed we already got the helper method 'assertQuery:opensNewWindow'
in ALGTest. The tests for the class methods stuff are now based on this.

TheGrayStone reviewed (approved):
Looks all good to me now!
